catroid/src/main/java/org/catrobat/catroid/content/Scene.java

```diff
@@ -172,8 +173,14 @@ public void removeClonedSprites() {
 	}
 
 	public void updateUserDataReferences(String oldName, String newName, UserData<?> item) {
-		for (Sprite sprite : spriteList) {
-			sprite.updateUserDataReferences(oldName, newName, item);
+		Project currentProject = ProjectManager.getInstance().getCurrentProject();
+		if (currentProject.getUserVariables().contains(item) || currentProject.getUserLists().contains(item)
+				|| currentProject.getMultiplayerVariables().contains(item)) {
```

Comment:
Maybe you could make this condition more readable if you outsource it to a function, e.g. *isGlobalVariable(UserData<?> item)*